doc/tutorial/conv_arithmetic.txt

```diff
@@ -0,0 +1,834 @@
+convolution with which it shares the kernel, and defining the equivalent direct
+convolution.
+
+No zero padding, unit strides, transposed
+-----------------------------------------
+
+The simplest way to think about a transposed convolution is by computing the
+output shape of the direct convolution for a given input shape first, and then
+inverting the input and output shapes for the transposed convolution.
+
+Let's consider the convolution of a :math:`3 \times 3` kernel on a :math:`4
+\times 4` input with unitary stride and no padding (i.e., :math:`i = 4`,
+:math:`k = 3`, :math:`s = 1` and :math:`p = 0`). As depicted in the convolution
+above, this produces a :math:`2 \times 2` output.  The transpose of this
+convolution will then have an output of shape :math:`4 \times 4` when applied on
+a :math:`2 \times 2` input.
+
+Another way to obtain the result of a transposed convolution is to apply an
+equivalent -- but much less efficient -- direct convolution. The example
+described so far could be tackled by convolving a :math:`3 \times 3` kernel over
+a :math:`2 \times 2` input padded with a :math:`2 \times 2` border of zeros
+using unit strides (i.e., :math:`i' = 2`, :math:`k' = k`, :math:`s' = 1` and
+:math:`p' = 2`), as shown here:
+
+.. figure:: conv_arithmetic_figures/no_padding_no_strides_transposed.gif
+
+Notably, the kernel's and stride's sizes remain the same, but the input of the
+transposed convolution is now zero padded.
+
+.. note::
+
+    Although equivalent to applying the transposed matrix, this visualization
+    adds a lot of zero multiplications in the form of zero padding. This is done
+    here for illustration purposes, but it is inefficient, and software
+    implementations will normally not perform the useless zero multiplications.
+
+One way to understand the logic behind zero padding is to consider the
+connectivity pattern of the transposed convolution and use it to guide the
+design of the equivalent convolution. For example, the top left pixel of the
+input of the direct convolution only contribute to the top left pixel of the
+output, the top right pixel is only connected to the top right output pixel,
+and so on.
+
+To maintain the same connectivity pattern in the equivalent convolution it is
+necessary to zero pad the input in such a way that the first (top-left)
+application of the kernel only touches the top-left pixel, i.e., the padding
+has to be equal to the size of the kernel minus one.
+
+Proceeding in the same fashion it is possible to determine similar observations
+for the other elements of the image, giving rise to the following relationship:
+
+.. admonition:: Relationship 7
+
+    A convolution described by :math:`s = 1`, :math:`p = 0` and :math:`k` has an
+    associated transposed convolution described by :math:`k' = k`, :math:`s' =
+    s` and :math:`p' = k - 1` and its output size is
+
+    .. math::
+
+        o' = i' + (k - 1).
```

Comment:
We use `i'` for the "equivalent direct convolution on the padded input" before. I think we should try to make it as clear as possible when we refer to one or the other. I suggest we use `o^T` and `i^T` to this purpose. This would also be consistent with the `C^T` notation. To make sure it's not misread as a transpose of a matrix, we can maybe add a note at the beginning of the transposed convolution section to introduce the notation.